packages/core/pattern.mjs

```diff
@@ -1368,9 +1372,32 @@ export function register(name, func) {
       return func(...args, pat);
     };
     mapFn = curry(mapFn, null, arity - 1);
-    return right.reduce((acc, p) => acc.appLeft(p), left.fmap(mapFn)).innerJoin();
+
+    // Don't use applicative for arguments that a) have a '__compose' function and b) are
+    // marked as being a higher order function parameter
+    function app(acc, p, i) {
+      if (f_params != null && f_params.length > i + 1 && f_params[i + 1] && '__compose' in p) {
```

Comment:
maybe things would be easier to define if f_params would be an array of indices? like `register('xx', .., [1]) ` tells that the argument with index 1 is higher order.

Reply:
I wondered about this too but think there will be other things we want to say about the parameters, like the default alignments ref #380.

Reply:
> I wondered about this too but think there will be other things we want to say about the parameters, like the default alignments ref #380.

I mean in that case a list of objects would probably be best

Reply:
jep.. makes sense